server/core/db/boltdb.go

```diff
@@ -134,13 +199,61 @@ func (b *BoltDB) Unlock(p models.Project, workspace string) (*models.ProjectLock
 			}
 			foundLock = true
 		}
-		return bucket.Delete([]byte(key))
+
+		if err := bucket.Delete([]byte(key)); err != nil {
+			return errors.Wrap(err, "failed to delete key")
+		}
+
+		// Dequeue next item
+		if updateQueue {
+			queueBucket := tx.Bucket(b.queueBucketName)
```

Comment:
The logic from lines 209 to 246 could be in its own function as well so it can be more easily unit tested.

Reply:
Moved to `(b *BoltDB) dequeue` with c04c4ce. However, I couldn't add add unit tests because the function is not exported (and the test is in a different package so it's not visible). Would you suggest to export the function so it can be unit tested?